### ISSUE TITLE
SQLitePlugin: "attempt to write a readonly database"

### ISSUE THREAD

**Issue by yankustefan** [ionic-team/ionic-view-issues#49]
When I try to run SQLitePlugin on ionic view, I get the following error: "attempt to write a readonly database"

running with "ionic run android --livereload" via Genymotion works fine.

Any clues?

**Comment by tlancina** [comment]
What's the database path you're trying to write to?

**Comment by yankustefan** [comment]
Hi tlancina,
Thank you for your fast feedback!

My current openDatabase function calls:

> `$window.sqlitePlugin.openDatabase({name: "ms_v139.db", location: 2, bgType: 1});`

I tried all "location" types and "bgType".

Do I also have to make a setting in config.xml?
At the moment I am not touching the preferences because I assume the CLI is making changes to it automatically.

**Comment by yankustefan** [comment]
If I drop bgType alltogether and set location to "0" it works.

> `$window.sqlitePlugin.openDatabase({name: NAME, location: 0});`

**Comment by yankustefan** [comment]
I still get "attempt to write a readonly database" after I synch the app to latest.
Unless I rename the db name or re-install Ionic View.

**Comment by ryandraper** [comment]
@yankustefan I am currently experiencing the exact same issue.  I changed to your settings from you Mar 20 comment and it fixed the issue, then later on the issue was back, but on somebody else's iPad, never mine :|
Any updates to this?

Did you notice if it happened after adding new columns to the database tables?
I wonder if it has anything to do with iCloud...

**Comment by mysimpex** [comment]
I've been trying to resolve this issue for a few days now and I start to catch on the problem. This issue happens when running the app in Ionic View on iOS only. 

I browsed through a lot of posts in regards to SQLite and initially I thought the issue is due to the database plugin in the Ionic View in iOS isn't ready when I attempt to access the database. But after setting up infinite recursive calls to check if the database plugin is ready, and infinite recursive calls to run "Create", "Delete", "Insert" and "Select" to check the accessibility of the database, I found that only iOS generates the error "attempt to write a readonly database" when I tries to access the database after I restart the app after the first execution.

During the first time execute, SQLite opens a new database with a new database name. All SQL scripts are executed without any error. On the second execution, the same database name is used, and SQLite is able to open and probably select (haven't tested this yet), but it definitely cannot delete or update the database.

Has any one discovered a solution for this issue?
